src/generic/datavgen.cpp

```diff
@@ -2394,15 +2394,17 @@ void wxDataViewMainWindow::OnPaint( wxPaintEvent &WXUNUSED(event) )
 
                 if ( m_useCellFocus && m_currentCol && m_currentColSetByKeyboard )
                 {
-                    renderColumnFocus = true;
-
-                    // If this is container node without columns, render full-row focus:
-                    if ( !IsList() )
+                    // If this is an item with a single column, render full-row focus:
+                    numColsWithValue = 0;
+                    unsigned int cols = GetOwner()->GetColumnCount();
+                    for ( unsigned int i = 0; i < cols; i++ )
                     {
-                        wxDataViewTreeNode *node = GetTreeNodeByRow(item);
-                        if ( node->HasChildren() && !model->HasContainerColumns(node->GetItem()) )
-                            renderColumnFocus = false;
+                        if ( model->HasValue(item, i) )
+                            numColsWithValue++;
+                        if ( numColsWithValue > 1 )
+                            break;
                     }
+                    renderColumnFocus = numColsWithValue<2;
```

Comment:
I'd just replace this with
```suggestion
                        if ( numColsWithValue > 1 )
                        {
                            renderColumnFocus = false;
                            break;
                        }
                    }
```
IMHO it's a bit clearer.

Reply:
By creating a function `HasItemMultipleValues` I have moved the new code out of here and left the logic here virtually identical to the original one. See commit 20243534ca9853a2d13a678379fe0a46398f01cf